File: src/Muflone/Factories/IDomainEventHandlerFactoryAsync.cs
using Muflone.Messages.Events;
using System.Collections.Generic;

namespace Muflone.Factories;

public interface IDomainEventHandlerFactoryAsync
{
	IEnumerable<IDomainEventHandlerAsync<T>> CreateDomainEventHandlersAsync<T>() where T : class, IDomainEvent;
}

File: php/src/Muflone/Factories/IDomainEventHandlerFactoryAsync.php
<?php

declare(strict_types=1);

namespace Muflone\Factories;

use Muflone\Messages\Events\IDomainEvent;
use Muflone\Messages\Events\IDomainEventHandlerAsync;

interface IDomainEventHandlerFactoryAsync
{
    /**
     * Create domain event handlers for the specified event type
     *
     * @template T of IDomainEvent
     * @param class-string<T> $eventClass
     * @return array<IDomainEventHandlerAsync>
     */
    public function createDomainEventHandlers(string $eventClass): array;
}
